### PR TITLE
After testing with UFS/GFS/FV-3, some tuning knob changes to Thompson-MP and icloud3 (cloud fraction) scheme

### PR DESCRIPTION
**Pull request by gthompsnWRF**
TYPE: enhancement

KEYWORDS: microphysics, clouds, cloud fraction, Thompson microphysics

SOURCE:  Greg Thompson (UCAR/Joint Center for Satellite Data Assimilation)

DESCRIPTION OF CHANGES:
The purpose of these changes is to produce better cloud coverage and radiation amounts per tuning done with 
GFS global model and comparisons to observations.

Problem:
The Thompson-MP code (from ccpp-physics) has been used within UFS/GFS/FV-3 global model for multiple 
day simulations.  Anning Cheng and Ruiyu Sun (NCEP/EMC) have performed most of the simulations and 
analysis and iterated with me to make adjustments to the Thompson-MP.  At first, EMC found that high 
clouds were not present enough, particularly in the tropics.  Once switching to a larger upper limit of cloud 
ice before it becomes snow, the cloud amounts improved but some clouds and radiation balances were still 
not matching results from FV3's GFDL microphysics (which had already been tuned to observations).  
Therefore an effort to account for subgrid-scale clouds using the cloud fraction scheme (icloud=3) was 
adopted to GFS and tested with various parameter settings.

Solution:
The biggest change is in how cloud ice converts to snow at a threshold size of 300 microns (up from 200 
microns) and for rimed snow to convert to graupel (changed from 250 to 350 microns).  A change to allowed 
max size of ice means that the fall velocity constant for ice was changed to keep it aligned with snow at the 
same cut-over size (of 300 microns).  In addition, the cloud fraction scheme (icloud=3) can further improve 
the clouds and radiation together with Thompson-MP due to the overall under-prediction of clouds.  The 
icloud3 option was also making far too many clouds when compared to observations so its tuning knobs 
were adjusted until attaining the following overall improvements compared to cloud and radiation global 
climatologies that EMC uses:  cloud amounts of low, middle, high, and total cloud coverage, longwave 
radiation outgoing at top-of-atmosphere, and shortwave radiation reaching the ground.

ISSUE:  
There are corresponding issues or pull requests in the ccpp-physics repo, 
[778](https://github.com/NCAR/ccpp-physics/pull/778)
[781](https://github.com/NCAR/ccpp-physics/pull/781)
[809](https://github.com/NCAR/ccpp-physics/pull/809)

LIST OF MODIFIED FILES: 
M  module_mp_thompson.F
M  module_radiation_driver.F
M  module_ra_rrtmg_lw.F
M  module_ra_rrtmg_lwf.F
M  module_ra_rrtmg_sw.F
M  module_ra_rrtmg_swf.F

TESTS CONDUCTED: 
1. A series of tests in GFS including 5, 7, 16, and 30-day long simulations compared to known cloud and 
radiation climatologies. The plots attached here in this comment were created by Anning Cheng. The numbers 
at the top of each panel represent global 3-day (days 3, 4, 5) average.

The first plot is the high/mid/low/total cloud amount.  The tunings in this PR reduced the high cloud amount 
from over 55% down to the low 40s, which matches observations of global cloud coverage pretty well.  And, 
another comparison is without the cloud fraction scheme, the mid-level clouds (and low/high) are simply less 
than observations, which is something well published in IPCC reports of global model clouds, especially the 
Southern Ocean and east sides of ocean basins (low stratus).
gfnl_cf_20190611

The next plot's panel (a) shows the outgoing longwave radiation.  The target value global average is about 
240 W/m2.  Without the cloud fraction scheme and changes to ```D0s``` and ```D0g```, the result would be 
too much outgoing longwave since it will not contain enough clouds at lower temperatures.  Also, the older 
version of the cloud fraction scheme with its excessive amount of high clouds would make the outgoing 
radiation closer to 225 W/m2.  So the tuning of the scheme brings the results closer to observations.

gfnl_prcp_20190611

Lastly, the final plot shows the downward longwave reaching the ground (panel c) and downward shortwave 
reaching the ground (panel d). 

gfnl_sfc_SWLW_20190611

2. Jenkins tests are passing.

RELEASE NOTE: Update of the Thompson microphysics scheme and cloud fraction scheme (icloud=3) to match the observations better. The modifications include updates to RRTMG LW and SW, and RRTMG fast LW and SW.

### CODE REVIEW

**Comment by davegill** [comment]
@gthompsnWRF 
Greg,
Would you add your affiliation please.

**Comment by davegill** [comment]
@weiwangncar @dudhia @gthompsnWRF 
Folks,
These are mostly just some tweaks to constants. Typical of Greg, this is a very tidy set of mods to review.

**Comment by davegill** [comment]
@dudhia @weiwangncar 
So, do we want this to go to v4.3.3 or to the develop branch for eventual v4.4. The mods do seem small enough for the v4.3.3 branch

**Comment by weiwangncar** [comment]
@davegill I'd say this should go to 4.4. It could potentially change the results quite a bit.
@gthompsnWRF Do you have any plots to show the differences, or impact of the change?

**Comment by gthompsnWRF** [comment]
> @gthompsnWRF Greg, Would you add your affiliation please.

got it

**Comment by gthompsnWRF** [comment]
> @davegill I'd say this should go to 4.4. It could potentially change the results quite a bit. @gthompsnWRF Do you have any plots to show the differences, or impact of the change?


I think Wei is right in putting this into v4.4.  Results will change for sure - that's pretty much the point to get better cloud prediction so the changes of the ```D0s``` variable alone were far larger than I had anticipated for sure.  Credit is due here to Ruiyu Sun for trying this when I was skeptical it would be so impactful.

@weiwangncar:  I can put some snapshots of results done with GFS model since those simulations were done for me by EMC, otherwise, I would have to run a before/after simulation with WRF to show the increase in upper level cloud ice.  Those listed PRs do have a lot of graphics and even full powerpoint files with tons of graphics that I created while testing things.  Those led up to these changes.  Also, I would need to get the most recent final graphics when changing the cloud fraction scheme through numerous tuning knob changes to get results to match the targets of the variables I listed in description.

**Comment by weiwangncar** [comment]
@gthompsnWRF Some plots from GFS tests are ok for us to get a glance at what kind of change is expected. Thanks.

**Comment by gthompsnWRF** [comment]
The plots attached here in this comment were created by Anning Cheng.   The numbers at the top of each panel represent global 3-day (days 3, 4, 5) average.

The first plot is the high/mid/low/total cloud amount.  The tunings in this PR reduced the high cloud amount from over 55% down to the low 40s, which matches observations of global cloud coverage pretty well.  And, another comparison is without the cloud fraction scheme, the mid-level clouds (and low/high) are simply less than observations, which is something well published in IPCC reports of global model clouds, especially the Southern Ocean and east sides of ocean basins (low stratus).
gfnl_cf_20190611

The next plot's panel (a) shows the outgoing longwave radiation.  The target value global average is about 240 W/m2.  Without the cloud fraction scheme and changes to ```D0s``` and ```D0g```, the result would be too much outgoing longwave since it will not contain enough clouds at lower temperatures.  Also, the older version of the cloud fraction scheme with its excessive amount of high clouds would make the outgoing radiation closer to 225 W/m2.  So the tuning of the scheme brings the results closer to observations.

gfnl_prcp_20190611

Lastly, the final plot shows the downward longwave reaching the ground (panel c) and downward shortwave reaching the ground (panel d).  I will fill in the global average target numbers when Anning lets me know those numbers.
gfnl_sfc_SWLW_20190611

**Comment by dudhia** [comment]
@gthompsnWRF I think for global radiation budgets the main constraints are outgoing longwave and shortwave which should be about 240 W/m2 and 100 W/m2 for annual global climatology.

File: phys/module_ra_rrtmg_sw.F [hunk 1]
```diff
@@ -10849,9 +10849,11 @@ SUBROUTINE RRTMG_SWRAD(                                        &
 
            if(iceflgsw.eq.5)then
               do k = kts, kte
-                 snow_mass_factor = 1.0                 
+                 snow_mass_factor = 0.99        ! Assume 1% of snow overlaps the cloud ice category
+                 gicewp = gicewp + (qs1d(k)*(1.0-snow_mass_factor) * pdel(ncol,k)*100.0 / gravmks * 1000.0)
                  if (resnow1d(ncol,k) .gt. 130.)then 
-                     snow_mass_factor = (130.0/resnow1d(ncol,k))*(130.0/resnow1d(ncol,k))
+                     snow_mass_factor = MIN(snow_mass_factor,                       &
+     &                         (130.0/resnow1d(ncol,k))*(130.0/resnow1d(ncol,k)))
```
Comment (davegill) [inline phys/module_ra_rrtmg_sw.F:10856]: @gthompsnWRF @weiwangncar @dudhia 
Greg,
With a cursory glance, it looks like the identical mods could go into RRTMG_SW fast.

**Comment by davegill** [comment]
@weiwangncar @dudhia 
Folks,
I added in the changes to RRTMG fast for LW and SW.

**Review by weiwangncar — APPROVED** [review]
Will see how the long simulation tests will show.

**Comment by gthompsnWRF** [comment]
> @weiwangncar @dudhia Folks, I added in the changes to RRTMG fast for LW and SW.

Thanks for doing this Dave - I was going to circle back around to it, just didn't have a chance yet.  Yes, it is best to include same mod in the 'fast' copies of RRTMG.  And this value of 0.99 can easily be tuned further if sensitivity testing shows it worthwhile.  Mike Iacono and I are working on a NOAA-JTTI project together where this could come into play.

**Comment by gthompsnWRF** [comment]
> Will see how the long simulation tests will show.

Please keep me posted.  Anning Cheng has now run with these code mods for 35 day long global simulation, BTW.

**Pull request merged**